common/firmware/i2c/i2c.c

```diff
@@ -0,0 +1,66 @@
+#include "common/firmware/i2c.h"
```

Comment:
I think this should probably be left up to the individual executables, or at least allow them to customize things like pins, etc.

Reply:
The same reasons that cause the HAL to have this kind of apply here - this code should be general and do only a minimal amount of autoconfiguration.